sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -89,20 +85,23 @@ class ResolveSessionCatalog(
             throw new AnalysisException("" +
               "ALTER COLUMN ... FIRST | ALTER is only supported with v2 tables.")
           }
+
+          val colName = a.column(0)
+          val dataType = a.dataType.getOrElse(v1Table.schema(colName).dataType)
```

Comment:
shall we consider case sensitivity?

Reply:
Sure. I noticed that we are not considering case sensitivities in v2 tables either: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala#L432

I can do a follow up PR if needed.

Reply:
Sure, will add.

Reply:
(btw, good catch!, it was a bug not considering case sensitivity)